packages/scene-composer/src/components/three-fiber/controls/PointerLockControls.tsx

```diff
@@ -0,0 +1,189 @@
+import * as THREE from 'three';
+
+import { PointerLockControls as PointerLockControlsImpl } from '../../../three/PointerLockControls';
+
+export type PointerLockControlsProps = ReactThreeFiber.Object3DNode<
+  PointerLockControlsImpl,
+  typeof PointerLockControlsImpl
+> & {
+  domElement?: HTMLElement;
+  selector?: string;
+  enabled?: boolean;
+  camera?: THREE.Camera;
+  onChange?: (e?: THREE.Event) => void;
+  onLock?: (e?: THREE.Event) => void;
+  onUnlock?: (e?: THREE.Event) => void;
+  makeDefault?: boolean;
+};
+
+export const PointerLockControls = forwardRef<PointerLockControlsImpl, PointerLockControlsProps>(
+  ({ domElement, selector, onChange, onLock, onUnlock, enabled = true, makeDefault, ...props }, ref) => {
+    const { camera, ...rest } = props;
+    const setEvents = useThree((state) => state.setEvents);
+    const gl = useThree((state) => state.gl);
+    const defaultCamera = useThree((state) => state.camera);
+    const invalidate = useThree((state) => state.invalidate);
+    const events = useThree((state) => state.events) as EventManager<HTMLElement>;
+    const get = useThree((state) => state.get);
+    const set = useThree((state) => state.set);
+    const explCamera = camera || defaultCamera;
+    const explDomElement = (domElement || events.connected || gl.domElement) as HTMLElement;
+
+    const controls = useMemo(() => new PointerLockControlsImpl(explCamera), [explCamera]);
+
+    const moveForward = useRef(false);
+    const moveBackward = useRef(false);
+    const moveLeft = useRef(false);
+    const moveRight = useRef(false);
+    const moveUp = useRef(false);
+    const moveDown = useRef(false);
+
+    const velocity = useRef(new THREE.Vector3());
+
+    useFrame((state, delta) => {
+      if (enabled && controls.isLocked) {
+        const newVelocity = velocity.current;
+        newVelocity.x -= velocity.current.x * 10.0 * delta;
+        newVelocity.y -= velocity.current.y * 10.0 * delta; //fly not fall
+        newVelocity.z -= velocity.current.z * 10.0 * delta;
+
+        const direction = new THREE.Vector3();
+        direction.x = Number(moveRight.current) - Number(moveLeft.current);
+        direction.y = Number(moveUp.current) - Number(moveDown.current);
+        direction.z = Number(moveForward.current) - Number(moveBackward.current);
+        direction.normalize(); // this ensures consistent movements in all directions
+
+        if (moveLeft.current || moveRight.current) newVelocity.x -= direction.x * 400.0 * delta;
+        if (moveUp.current || moveDown.current) newVelocity.y -= direction.y * 400.0 * delta;
+        if (moveForward.current || moveBackward.current) newVelocity.z -= direction.z * 400.0 * delta;
+
+        controls.moveRight(-newVelocity.x * delta);
+        controls.moveForward(-newVelocity.z * delta);
+        controls.moveUp(-newVelocity.y * delta);
+
+        velocity.current = newVelocity;
+      }
+    });
+
+    const onKeyDown = useCallback((event: KeyboardEvent) => {
```

Comment:
Is there a reason that this isn't setup as a reusable util? We need to make sure that any keyboard management we're introducing is consistent & has no collisions with other features.

Reply:
It's kind of straight port of the example code from three-stblib and drei.... which is what we also did for the other camera controls. We had a design doc a while back about doing a key binding control utility like your talking about but it didn't get implemented yet.

Reply:
Done a basic stripped done version of the the key mapping that we'll revisit later.